bpf/lib/nodeport.h

```diff
@@ -2081,6 +2081,45 @@ nodeport_extract_dsr_v4(struct __ctx_buff *ctx,
 	return 0;
 }
 
+/* remove_dsr_ip_opt_v4 removes the DSR IP Option from the packet and 
+ * recalculates the IPv4 checksum.
+ */
+static __always_inline int
+remove_dsr_ip_opt_v4(struct __ctx_buff *ctx, struct iphdr *ip4) {
+	struct dsr_opt_v4 opt;
+    __u32 iph_old, iph_new, sum_l3;
+
+	/* Check for DSR IP Option */
+	if (ip4->ihl < 0x7)
+		return CTX_ACT_OK;
+	if (ctx_load_bytes(ctx, ETH_HLEN + sizeof(struct iphdr), &opt, sizeof(opt)) < 0)
+		return DROP_INVALID;
+	if (opt.type != DSR_IPV4_OPT_TYPE)
+		return CTX_ACT_OK;
+
+	iph_old = *(__u32 *)ip4;
+	/*  Remove option */
+	ip4->ihl -= sizeof(opt) >> 2;
+    ip4->tot_len = bpf_htons(bpf_ntohs(ip4->tot_len) - sizeof(opt));
```

Comment:
indentation is inconsistent here, also on line 2090 above

Reply:
Thanks; done.